src/barril/units/_array.py

```diff
@@ -31,45 +49,75 @@ class Array(AbstractValueWithQuantityObject):
         Array(ObtainQuantity("m", "length"), [0, 1, 2, 3, 4])
     """
 
-    def __init__(self, category, values=None, unit=None):
-        AbstractValueWithQuantityObject.__init__(self, category, value=values, unit=unit)
-
-    def _InternalCreateWithQuantity(self, quantity, values=None, unit_database=None, value=None):
+    @overload
```

Comment:
That's a lot of options for init hehe.

Reply:
Yep, here overload really helps to unravel the hidden options we have.